src/plots/cartesian/set_convert.js

```diff
@@ -152,10 +152,8 @@ module.exports = function setConvert(ax, fullLayout) {
         if(ax._categoriesMap) {
             index = ax._categoriesMap[v];
             if(index !== undefined) return index;
-        } else {
-            index = ax._categories.indexOf(v);
-            if(index !== -1) return index;
```

Comment:
Oh sorry, I just meant we can remove the fallbacks for if `_categoriesMap` doesn't even exist, ie in `getCategoryIndex` changing:
```js
if(ax._categoriesMap) {
    var index = ax._categoriesMap[v];
    if(index !== undefined) return index;
}
```
to just:
```js
var index = ax._categoriesMap[v];
if(index !== undefined) return index;
```
and removing from `setCategoryIndex`:
```js
if(ax._categoriesMap === undefined) {
    ax._categoriesMap = {};
}
```

As far as I can tell, *that* much should be robust. Not a huge deal but should be a small performance boost.

But you raise a good point about future changes, I guess that's an argument for converting these two to a single bidirectional map object even if it doesn't yield any additional perf benefits.